internal/api/token_refresh.go

```diff
@@ -64,31 +64,18 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		}
 
 		if session != nil {
-			var notAfter time.Time
+			result := session.CheckValidity(retryStart, &token.UpdatedAt, config.Sessions.Timebox, config.Sessions.InactivityTimeout)
 
-			if session.NotAfter != nil {
-				notAfter = *session.NotAfter
-			}
-
-			if config.Sessions.Timebox != nil {
-				sessionEndsAt := session.CreatedAt.Add((*config.Sessions.Timebox).Abs())
+			switch result {
+			case models.SessionValid:
+				// do nothing
 
-				if notAfter.IsZero() || notAfter.After(sessionEndsAt) {
-					notAfter = sessionEndsAt
-				}
-			}
+			case models.SessionTimedOut:
+				return oauthError("invalid_grant", "Invalid Refresh Token: Session Expired (Inactivity)")
 
-			if !notAfter.IsZero() && a.Now().After(notAfter) {
+			default:
```

Comment:
Confirming my understanding here-  we merge the `SessionPastTimebox` and `SessionNotAfter` cases since they're an internal concept and the dev won't need to distinguish between the two cases so long as they know if it times out?

Reply:
Yeah, the session has expired -- it doesn't matter in what way.